tools/release/generate_release_notes.rb

```diff
@@ -0,0 +1,107 @@
+
+report << "[[logstash-#{current_release_dashes}]]"
+report << "=== Logstash #{current_release} Release Notes\n"
+
+plugin_changes = {}
+
+report <<  "---------- DELETE FROM HERE ------------"
+report <<  "=== Logstash Pull Requests with label v#{current_release}\n"
+
+uri = URI.parse("https://api.github.com/search/issues?q=repo:elastic/logstash+is:pr+is:closed+label:v#{current_release}&sort=created&order=asc")
+pull_requests = JSON.parse(Net::HTTP.get(uri))
+pull_requests['items'].each do |prs|
+  report << "* #{prs['title']} #{prs['html_url']}[##{prs['number']}]"
+end
+report << ""
+
+report <<  "=== Logstash Commits between #{release_branch} and #{previous_release_tag}\n"
+report <<  "Computed with \"git log --pretty=format:'%h -%d %s (%cr) <%an>' --abbrev-commit --date=relative v#{previous_release_tag}..#{release_branch}\""
+report <<  ""
+logstash_prs =  `git log --pretty=format:'%h -%d %s (%cr) <%an>' --abbrev-commit --date=relative v#{previous_release_tag}..#{release_branch}`
+report <<  logstash_prs
+report << "\n=== Logstash Plugin Release Changelogs ==="
+report << "Computed from \"git diff v#{previous_release_tag}..#{release_branch} *.release\""
+result = `git diff v#{previous_release_tag}..#{release_branch} *.release`.split("\n")
+
+result.each do |line|
+  # example "+    logstash-input-syslog (3.4.1)"
+  if match = line.match(/\+\s+(?<plugin>logstash-.+?-.+?)\s+\((?<version>\d+\.\d+.\d+).*?\)/)
+    plugin_changes[match[:plugin]] ||= []
+    plugin_changes[match[:plugin]] << match[:version]
+  elsif match = line.match(/\-\s+(?<plugin>logstash-.+?-.+?)\s+\((?<version>\d+\.\d+.\d+).*?\)/)
+    plugin_changes[match[:plugin]] ||= []
+    plugin_changes[match[:plugin]].unshift(match[:version])
+  else
+    # ..
+  end
+end
+report << "Changed plugin versions:"
+plugin_changes.each {|p, v| report << "#{p}: #{v.first} -> #{v.last}" }
+report << "---------- DELETE UP TO HERE ------------\n"
+
+report << "==== Plugins\n"
+
+plugin_changes.each do |plugin, versions|
+  _, type, name = plugin.split("-")
+  header = "*#{name.capitalize} #{type.capitalize}*"
+  start_changelog_file = Tempfile.new(plugin + 'start')
+  end_changelog_file = Tempfile.new(plugin + 'end')
+  changelog = `curl https://raw.githubusercontent.com/logstash-plugins/#{plugin}/v#{versions.last}/CHANGELOG.md`.split("\n")
+  report << "#{header}\n"
+  changelog.each do |line|
+    break if line.match(/^## #{versions.first}/)
+    next if line.match(/^##/)
+    line.gsub!(/^\+/, "")
+    line.gsub!(/ #(?<number>\d+)\s*$/, " https://github.com/logstash-plugins/#{plugin}/issues/\\k<number>[#\\k<number>]")
+    line.gsub!(/^\s+-/, "*")
+    report << line
+  end
+  report << ""
+  start_changelog_file.unlink
+  end_changelog_file.unlink
+end
+
+release_notes.insert(release_notes_entry_index, report.join("\n").gsub(/\n{3,}/, "\n\n"))
+
+IO.write(RELEASE_NOTES_PATH, release_notes.join("\n"))
+
+puts "Creating commit.."
```

Comment:
same comment here, should we add a validation step to make sure we want to proceed with the commit and push?